scrapegraphai/utils/proxy_rotation.py

```diff
@@ -161,7 +161,7 @@ def _search_proxy(proxy: Proxy) -> ProxySettings:
     Returns:
         A 'playwright' compliant proxy configuration.
     """
-    server = search_proxy_servers(max_shape=1, **proxy.get("criteria", {}))[0]
+    server = search_proxy_servers(**proxy.get("criteria", {}))[0]
```

Comment:
only reason why I had forced `max_shape=1` is cause that function will only be called by the fetch node; proxy-seeking users might as well call `search_proxy_servers` with the desired parameters.

`_search_proxy` is called only by `parse_or_search_proxy` which has been designed for the fetch node and specifically looks to generate one single proxy server satisfying criteria (fetch node won't use more than one)

Reply:
I see, I had removed it since it was conflicting with the max_shape specified by the user if present. I have added it back now removing the user specified max_shape if present. will make a new merge